src/mtev_cluster.c

```diff
@@ -236,22 +268,52 @@ mtev_cluster_info_compose(void *payload, int len, void *c) {
   packed_time |= (my_boot_time.tv_usec & 0xffffff); /* 3 bytes */
   packed_time = htonll(packed_time);
   MEMWRITE(&packed_time, sizeof(packed_time));
-  seq = htonll(cluster->config_seq);
-  MEMWRITE(&seq, sizeof(seq));
+  my_seq = htonll(cluster->config_seq);
+  MEMWRITE(&my_seq, sizeof(my_seq));
+
+  assert(HEART_BEAT_HDR_LEN == MEMWRITE_WRITTEN);
+
+  if(cluster->hb_payloads->length != 0) {
+    // write header (pointer table)
+    for(i=0; i != cluster->hb_payloads->length; ++i) {
+      hb_payload = jl_array_list_get_idx(cluster->hb_payloads, i);
+      MEMWRITE(&hb_payload->app_id, 1);
+      MEMWRITE(&hb_payload->key, 1);
+      MEMWRITE(&hb_payload->data_len, 1);
+    }
+    // write payload
+    for(i=0; i != cluster->hb_payloads->length; ++i) {
+      hb_payload = jl_array_list_get_idx(cluster->hb_payloads, i);
+      MEMWRITE(hb_payload->data, hb_payload->data_len);
+    }
+  }
 
   /* TODO: Support registered composers */
-  MEMWRITE_RETURN;
+  return MEMWRITE_WRITTEN;
+}
+static void
+mtev_cluster_store_payload(mtev_cluster_node_t *node, const void* payload, uint8_t payload_length) {
+  if(node->payload) {
+    free (node->payload);
+  }
+  node->payload = malloc(payload_length);
```

Comment:
You're absolutely right! Should be fixed now. Will be tested tomorrow with other changes to come